lib/Backends/NNPI/tests/TestBlackList.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) Glow Contributors. See CONTRIBUTORS file.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <cstring>
+#include <iostream>
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
+namespace TestBlackList {
+
+enum NNPI_DEVICE_VERSION {
+  NNPI_DEVICE_VERSION_1 = 0b001,
+  NNPI_DEVICE_VERSION_2 = 0b010,
+  NNPI_DEVICE_VERSION_3 = 0b100,
+  NNPI_DEVICE_VERSION_ANY = 0b111,
+};
+
+enum NNPI_EXECUTION_ENGINE {
+  NNPI_EXECUTION_ENGINE_SW = 0b01 << 16,
+  NNPI_EXECUTION_ENGINE_HW = 0b10 << 16,
+  NNPI_EXECUTION_ENGINE_ANY = 0b11 << 16,
+};
+
+const uint32_t AnyDeviceAnyEngine =
+    NNPI_DEVICE_VERSION_ANY | NNPI_EXECUTION_ENGINE_ANY;
+const uint32_t AnyDeviceEngineHW =
+    AnyDeviceAnyEngine ^ NNPI_EXECUTION_ENGINE_SW;
+const uint32_t AnyDeviceEngineSW =
+    AnyDeviceAnyEngine ^ NNPI_EXECUTION_ENGINE_HW;
```

Comment:
Naming here is a bit confusing -- "AnyDeviceAnyEngine" vs. "AnyDeviceEngineHW". I'd suggest "AnyDeviceHWEngine"